contracts/03-tamagotchi-nft/src/lib.rs

```diff
@@ -129,6 +131,32 @@ extern fn handle() {
                     msg::reply(TmgEvent::Slept, 1).expect("Error in a reply'tamagotchi::slept'");
                 }
             }
+            TmgAction::Transfer(account) => {
+                tmg.owner = account;
+                msg::reply(TmgEvent::Transferred(account), 0)
+                    .expect("Error in a reply'tamagotchi::transferred'");
+            }
+            TmgAction::Approve(account) => {
+                tmg.approved_account = Some(account);
+                msg::reply(TmgEvent::Approved(account), 0)
+                    .expect("Error in a reply'tamagotchi::approved'");
+            }
+            TmgAction::RevokeApproval => {
+                tmg.approved_account = None;
+                msg::reply(TmgEvent::ApprovalRevoked, 0)
+                    .expect("Error in a reply'tamagotchi::approval_revoked'");
+            }
+        }
+    } else if msg::source() == tmg.approved_account.unwrap_or_default() {
```

Comment:
Better would be better to check the account permissions inside the `TmgAction::Transfer` branch to not double the code.

Also, it is better to not call `msg::source` twice.

Reply:
<img width="700" alt="image" src="https://github.com/Jackliu-miaozi/Jackliu-gear-academy/assets/71253778/627bc2a6-0b26-485f-ba3c-76caf3dfd04f">

Reply:
<img width="694" alt="image" src="https://github.com/Jackliu-miaozi/Jackliu-gear-academy/assets/71253778/042d4077-24b4-481c-8807-0b3c0f420900">